src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -861,18 +897,23 @@ impl LinkCollector<'_, '_> {
         let cx = self.cx;
         let link = ori_link.replace("`", "");
         let parts = link.split('#').collect::<Vec<_>>();
-        let (link, extra_fragment) = if parts.len() > 2 {
+        if parts.len() > 2 {
+            // A valid link can't have multiple #'s
             anchor_failure(cx, &item, &link, dox, link_range, AnchorFailure::MultipleAnchors);
             return;
-        } else if parts.len() == 2 {
-            if parts[0].trim().is_empty() {
-                // This is an anchor to an element of the current page, nothing to do in here!
+        }
+
+        let link = parts[0];
+        let extra_fragment = if parts.len() == 2 {
+            if link.trim().is_empty() {
+                // This is an anchor to an element of the current page (#foo), nothing to do in here!
                 return;
             }
-            (parts[0], Some(parts[1].to_owned()))
+            Some(parts[1].to_owned())
```

Comment:
Doesn't need to be fixed here, but I wonder if this copy is actually necessary ...